libs/prelude/Prelude/Types.idr

```diff
@@ -490,13 +490,14 @@ Foldable List where
 
   foldMap f = foldl (\acc, elem => acc <+> f elem) neutral
 
+public export listBindOnto : (a -> List b) -> List b -> List a -> List b
+listBindOnto f xs []        = reverse xs
```

Comment:
It would be natural to use a `SnocList a` for the accumulator and then chip the result:
```suggestion
listBindOnto f xs []        = xs <>> []
```
but I think we might need to move more stuff into `Prelude` from `base`, so maybe it's fine to keep as lists.

Reply:
I think you mean to have `listBindOnto` return `SnocList` and use `<><`. Otherwise we iterate over the accumulator each time.

Reply:
I think we just to need to change the order, so `listBindOnto` is defined after fish and chip (which are both in this file as well).